### PR TITLE
chore(flask): flask 1.1.1

### PR DESCRIPTION
**Pull request by paulineribeyre**
Bump flask to the latest version
also bump pytest to 4.6.5 (latest version that supports python 2) and pytest-flask to 0.15.0 (latest version)

### Dependency updates
- Flask 1.1.1, Werkzeug 0.16.0

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
Jenkins Build 4 : time taken 42 min
Check the https://jenkins.planx-pla.net/job/Performance%20tests%20for%20uc-cdis/job/sheepdog/job/PR-300/4/display/redirect


Test results:
	Passed: 109, Failed: 0, Skipped: 0



| Test | Time (`PR`) | Time (`master`) | Diff |
| - | - | - | - |
| @ExportPerformanceTests @Performance: Export all nodes of type: aliquot | 0.382 | 0.482 | -0.10 |
| @ExportPerformanceTests @Performance: Export all nodes of type: case | 0.287 | 0.215 | 0.07 |
| @ExportPerformanceTests @Performance: Export all nodes of type: experiment | 0.597 | 0.337 | 0.26 |
| @ExportPerformanceTests @Performance: Export all nodes of type: program | 0.111 | 0.048 | 0.06 |
| @ExportPerformanceTests @Performance: Export all nodes of type: project | 0.049 | 0.054 | -0.01 |
| @ExportPerformanceTests @Performance: Export all nodes of type: read_group | 0.479 | 0.555 | -0.08 |
| @ExportPerformanceTests @Performance: Export all nodes of type: read_group_qc | 0.118 | 0.14 | -0.02 |
| @ExportPerformanceTests @Performance: Export all nodes of type: sample | 0.384 | 0.433 | -0.05 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: aliquot | 0.291 | 0.202 | 0.09 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: case | 0.39 | 0.415 | -0.03 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: experiment | 0.435 | 0.272 | 0.16 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: read_group | 0.154 | 0.084 | 0.07 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: read_group_qc | 0.058 | 0.08 | -0.02 |
| @ExportPerformanceTests @Performance: Exporting a record by ID on nodes of type: sample | 0.148 | 0.183 | -0.03 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 0 | 0.574 | 0.795 | -0.22 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 1 | 0.705 | 0.651 | 0.05 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 2 | 0.658 | 0.537 | 0.12 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 3 | 0.481 | 0.538 | -0.06 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 4 | 0.358 | 0.4 | -0.04 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 5 | 0.298 | 0.331 | -0.03 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 6 | 0.229 | 0.206 | 0.02 |
| @QueryPerformanceTests @Performance: Executing bottomUp query # 7 | 0.093 | 0.138 | -0.04 |
| @QueryPerformanceTests @Performance: Executing topDown query # 0 | 0.076 | 0.105 | -0.03 |
| @QueryPerformanceTests @Performance: Executing topDown query # 2 | 0.203 | 0.124 | 0.08 |
| @QueryPerformanceTests @Performance: Executing topDown query # 4 | 0.382 | 0.266 | 0.12 |
| @QueryPerformanceTests @Performance: Executing topDown query # 5 | 0.398 | 0.56 | -0.16 |
| @QueryPerformanceTests @Performance: Executing topDown query # 6 | 0.503 | 0.457 | 0.05 |
| @QueryPerformanceTests @Performance: Executing topDown query # 7 | 0.566 | 0.621 | -0.06 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 10 | 0.402 | 0.504 | -0.10 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 100 | 3.153 | 3.953 | -0.80 |
| @SubmissionPerformanceTests @Performance: Submission acknowledgement 1000 | 30.016 | 36.652 | -6.64 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 10 | 1.773 | 2.32 | -0.55 |
| @SubmissionPerformanceTests @Performance: Submission aligned_reads_index 100 | 16.907 | 19.51 | -2.60 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 10 | 0.44 | 0.504 | -0.06 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 100 | 3.153 | 3.482 | -0.33 |
| @SubmissionPerformanceTests @Performance: Submission aliquot 1000 | 30.465 | 40.726 | -10.26 |
| @SubmissionPerformanceTests @Performance: Submission case 10 | 0.485 | 0.517 | -0.03 |
| @SubmissionPerformanceTests @Performance: Submission case 100 | 3.318 | 3.944 | -0.63 |
| @SubmissionPerformanceTests @Performance: Submission case 1000 | 29.778 | 36.092 | -6.31 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 10 | 0.488 | 0.667 | -0.18 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 100 | 3.797 | 4.472 | -0.68 |
| @SubmissionPerformanceTests @Performance: Submission clinical_test 1000 | 33.245 | 40.815 | -7.57 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 10 | 0.583 | 0.546 | 0.04 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 100 | 3.072 | 3.613 | -0.54 |
| @SubmissionPerformanceTests @Performance: Submission core_metadata_collection 1000 | 29.895 | 35.829 | -5.93 |
| @SubmissionPerformanceTests @Performance: Submission demographic 10 | 0.442 | 0.543 | -0.10 |
| @SubmissionPerformanceTests @Performance: Submission demographic 100 | 3.211 | 4.234 | -1.02 |
| @SubmissionPerformanceTests @Performance: Submission demographic 1000 | 33.599 | 40.395 | -6.80 |
| @SubmissionPerformanceTests @Performance: Submission diagnosis 10 | 0.462 | 0.593 | -0.13 |
| @SubmissionPerformanceTests @Performance: Submission diagnosis 100 | 3.73 | 4.368 | -0.64 |
| @SubmissionPerformanceTests @Performance: Submission diagnosis 1000 | 39.156 | 43.96 | -4.80 |
| @SubmissionPerformanceTests @Performance: Submission experiment 10 | 0.598 | 0.641 | -0.04 |
| @SubmissionPerformanceTests @Performance: Submission experiment 100 | 3.305 | 3.651 | -0.35 |
| @SubmissionPerformanceTests @Performance: Submission experiment 1000 | 30.842 | 35.814 | -4.97 |
| @SubmissionPerformanceTests @Performance: Submission experimental_metadata 10 | 1.695 | 2.299 | -0.60 |
| @SubmissionPerformanceTests @Performance: Submission experimental_metadata 100 | 16.642 | 19.743 | -3.10 |
| @SubmissionPerformanceTests @Performance: Submission exposure 10 | 0.425 | 0.703 | -0.28 |
| @SubmissionPerformanceTests @Performance: Submission exposure 100 | 3.086 | 3.635 | -0.55 |
| @SubmissionPerformanceTests @Performance: Submission exposure 1000 | 30.234 | 36.11 | -5.88 |
| @SubmissionPerformanceTests @Performance: Submission family_history 10 | 0.421 | 0.528 | -0.11 |
| @SubmissionPerformanceTests @Performance: Submission family_history 100 | 3.078 | 3.509 | -0.43 |
| @SubmissionPerformanceTests @Performance: Submission family_history 1000 | 30.754 | 35.886 | -5.13 |
| @SubmissionPerformanceTests @Performance: Submission keyword 10 | 0.405 | 0.494 | -0.09 |
| @SubmissionPerformanceTests @Performance: Submission keyword 100 | 3.09 | 3.604 | -0.51 |
| @SubmissionPerformanceTests @Performance: Submission keyword 1000 | 29.311 | 38.68 | -9.37 |
| @SubmissionPerformanceTests @Performance: Submission publication 10 | 0.41 | 0.496 | -0.09 |
| @SubmissionPerformanceTests @Performance: Submission publication 1000 | 30.179 | 36.532 | -6.35 |
| @SubmissionPerformanceTests @Performance: Submission read_group 10 | 0.44 | 0.624 | -0.18 |
| @SubmissionPerformanceTests @Performance: Submission read_group 100 | 3.251 | 3.617 | -0.37 |
| @SubmissionPerformanceTests @Performance: Submission read_group 1000 | 32.943 | 42.742 | -9.80 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 10 | 0.594 | 0.712 | -0.12 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 100 | 4.935 | 5.55 | -0.62 |
| @SubmissionPerformanceTests @Performance: Submission read_group_qc 1000 | 34.438 | 41.761 | -7.32 |
| @SubmissionPerformanceTests @Performance: Submission sample 10 | 0.462 | 0.537 | -0.07 |
| @SubmissionPerformanceTests @Performance: Submission sample 1000 | 33.967 | 39.344 | -5.38 |
| @SubmissionPerformanceTests @Performance: Submission slide 10 | 0.416 | 0.532 | -0.12 |
| @SubmissionPerformanceTests @Performance: Submission slide 100 | 3.391 | 4.279 | -0.89 |
| @SubmissionPerformanceTests @Performance: Submission slide 1000 | 31.226 | 37.737 | -6.51 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 10 | 0.427 | 0.506 | -0.08 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 100 | 3.308 | 3.601 | -0.29 |
| @SubmissionPerformanceTests @Performance: Submission slide_count 1000 | 31.543 | 36.591 | -5.05 |
| @SubmissionPerformanceTests @Performance: Submission slide_image 10 | 1.992 | 2.293 | -0.30 |
| @SubmissionPerformanceTests @Performance: Submission slide_image 100 | 17.4 | 19.691 | -2.29 |
| @SubmissionPerformanceTests @Performance: Submission submitted_aligned_reads 10 | 1.83 | 2.2 | -0.37 |
| @SubmissionPerformanceTests @Performance: Submission submitted_aligned_reads 100 | 17.355 | 20.222 | -2.87 |
| @SubmissionPerformanceTests @Performance: Submission submitted_copy_number 10 | 1.691 | 1.976 | -0.28 |
| @SubmissionPerformanceTests @Performance: Submission submitted_copy_number 100 | 16.475 | 18.85 | -2.38 |
| @SubmissionPerformanceTests @Performance: Submission submitted_methylation 10 | 1.689 | 2.122 | -0.43 |
| @SubmissionPerformanceTests @Performance: Submission submitted_methylation 100 | 17.216 | 19.374 | -2.16 |
| @SubmissionPerformanceTests @Performance: Submission submitted_somatic_mutation 10 | 1.717 | 2.195 | -0.48 |
| @SubmissionPerformanceTests @Performance: Submission submitted_somatic_mutation 100 | 16.659 | 19.636 | -2.98 |
| @SubmissionPerformanceTests @Performance: Submission submitted_unaligned_reads 10 | 1.869 | 2.294 | -0.42 |
| @SubmissionPerformanceTests @Performance: Submission submitted_unaligned_reads 100 | 16.978 | 19.575 | -2.60 |
| @SubmissionPerformanceTests @Performance: Submission treatment 10 | 0.416 | 0.531 | -0.12 |
| @SubmissionPerformanceTests @Performance: Submission treatment 100 | 3.289 | 3.821 | -0.53 |
| @SubmissionPerformanceTests @Performance: Submission treatment 1000 | 32.384 | 38.853 | -6.47 |

File: sheepdog/blueprint/routes/views/program/project.py [hunk 1]
```diff
@@ -410,7 +410,7 @@ def export_entities(program, project):
     project_id = "{}-{}".format(program, project)
     file_format = kwargs.get("file_format") or "tsv"
 
-    mimetype = "application/octet-stream"
+    mimetype = "application/json" if file_format.lower() == "json" else "application/octet-stream"
```
Comment (Avantol13) [inline sheepdog/blueprint/routes/views/program/project.py]: why was this necessary?

File: tests/integration/datadictwithobjid/conftest.py [hunk 2]
```diff
@@ -36,7 +36,6 @@ def get_parent(path):
 )
 
 
-@pytest.fixture(scope="session")
```
Comment (Avantol13) [inline tests/integration/datadictwithobjid/conftest.py]: I assume this is now the default?

File: sheepdog/blueprint/routes/views/program/project.py (hunk 1 above)
Comment (paulineribeyre) [reply sheepdog/blueprint/routes/views/program/project.py]: the unit tests are testing that when you hit the export endpoint with `format=json`, `response.json` contains the data. For other formats, `response.data` contains the data

In the new Flask code, `response.json` is None if the mimetype is not `application/json`, unless you force it to parse the data into JSON by using `response.get_json(force=True)` instead of `response.json`

I think in the older versions, doing `response.json` was forcing it by default, and throwing an error if it wasn't JSON

File: tests/integration/datadictwithobjid/conftest.py (hunk 2 above)
Comment (paulineribeyre) [reply tests/integration/datadictwithobjid/conftest.py]: no, but the `pg_config` function is never used as a fixture, it's only used once as a function. The new version of pytest-flask protests that fixtures are not meant to be called
